docs/operating-scylla/security/authorization.rst

```diff
@@ -91,6 +91,30 @@ If the option is used and the role exists, the statement is a no-op::
     CREATE ROLE other_role;
     CREATE ROLE IF NOT EXISTS other_role;
 
+Creating a role with a hashed password
+``````````````````````````````````````
+
+When you create a role with a password, the password is hashed by ScyllaDB and only then inserted into the database.
+However, there also exists a version of the statement allowing you to create a role with a pre-hashed password; the syntax:
+
+.. code-block::
+
+   create_role_statement: CREATE ROLE [ IF NOT EXISTS ] `role_name`
+                        : WITH HASHED PASSWORD `hashed_password`
+                        :     ( AND `role_option` )*
+   hashed_password: `string`
+   role_option: LOGIN '=' `boolean`
+              :| SUPERUSER '=' `boolean`
+              :| OPTIONS '=' `map_literal`
+
+When that statement is executed, ScyllaDB will not hash the provided hashed password; it will be inserted into
+the database directly, with no additional processing of it. The statement only works with an authenticator that
+uses passwords, e.g. PasswordAuthenticator. It can only be executed by a role with the superuser privilege.
+
+The provided hashed password must be an encrypted string using the library function `crypt(3)` specified by POSIX,
+and it must use one of the encryption algorithms that ScyllaDB supports: bcrypt, sha512crypt, sha256crypt, md5crypt.
+Not abiding by those requirements may result in undefined behavior, so you should be very careful when executing the statement.
```

Comment:
"Undefined behavior" sounds very scary. What's the worst that can actually happen? The user won't be able to log in, or the whole database crashes?

If it's the latter then we should improve how we handle malformed hashed password.

Reply:
What happens now is we insert the hashed password into the database no matter what it is, but if it's invalid (i.e. it doesn't follow one of those algorithms), trying to log in will result in a system error—that's what I've observed. In that case, the best the user can do is remove the role from the database and create it again. I used the wording with undefined behaviour because there might be more side effects of that that I haven't thought of.